components/scream/src/physics/p3/p3_main_impl.hpp

```diff
@@ -799,12 +799,14 @@ ::p3_main_part3(
       get_cloud_dsd2(qc_incld, nc_incld, mu_c(k), rho(k), nu(k), dnu, lamc(k), ignore1, ignore2, qc_gt_small);
 
       nc(k).set(qc_gt_small,nc_incld*cld_frac_l(k)); //cld_dsd2 might have changed incld nc... need consistency.
-      diag_eff_radius_qc(k)       .set(qc_gt_small, sp(0.5) * (mu_c(k) + 3) / lamc(k));
-      qv(k)              .set(qc_small, qv(k)+qc(k));
-      th_atm(k)              .set(qc_small, th_atm(k)-inv_exner(k)*qc(k)*latent_heat_vapor(k)*inv_cp);
-      vap_liq_exchange(k).set(qc_small, vap_liq_exchange(k) - qc(k));
-      qc(k)              .set(qc_small, 0);
-      nc(k)              .set(qc_small, 0);
+      if (qc_gt_small.any()) {
```

Comment:
This code would be improved by one overall `if (qc_gt_small.any())`. That would then naturally account for the /0 check.